src/components/Settings/NewStudentForm.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState, FC } from 'react';
+
+const NewStudentForm: FC = () => {
+  const [studentData, setStudentData] = useState({
```

Comment:
Není dobrý držet values formuláru v useState. Při každé změne se re-renderuje celý komponent a to úplne nechceš. Lépe použit React hook forms nebo jiné forms řešení